src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -0,0 +1,55 @@
+import {useEffect} from 'react';
+import type {ComponentType} from 'react';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import useNetwork from '@hooks/useNetwork';
+import {openPolicyAccountingPage} from '@libs/actions/PolicyConnections';
+import withPolicy from './withPolicy';
+import type {WithPolicyProps} from './withPolicy';
+
+type WithPolicyConnectionsProps = WithPolicyProps;
+
+function withPolicyConnections(WrappedComponent: ComponentType<WithPolicyConnectionsProps>) {
+    function WithPolicyConnections({policy, policyMembers, policyDraft, policyMembersDraft, route}: WithPolicyConnectionsProps) {
+        const {isOffline} = useNetwork();
+
+        useEffect(() => {
+            if (!policy?.connections || policy?.connections) {
```

Comment:
How does this logic work? I read it as "if there are no connection OR if there are connections, then return early".

Could you please add a code comment here to explain the intention of the `useEffect()`? Be sure to mention the difference between active policies already having `connections` and non-active policies not having them.

Reply:
I added a comment in https://github.com/Expensify/App/pull/39132/commits/8e32fded9dff0a2401b310b4a31efeee4708d9d8

Reply:
> Be sure to mention the difference between active policies already having connections and non-active policies not having them.

Is it accurate to say that the active policy is the one selected in the Workspace Switcher or the one whose workspace settings are displayed? In other words, is the active policy the one whose ID is displayed in the URL?

Given this definition of an active policy, your comment above confuses me a bit. The withPolicyConnections HOC uses the withPolicy HOC in its implementation, which implies that the policy under consideration is always the active one.

Reply:
Hm, no... what I mean by "active policy" is the policy that is set to be the default policy on the OldDot Workspace page. This is the policy that is sent to NewDot in `OpenApp` and it should contain the connections.

Reply:
ah I apologize for my misunderstanding 🙇 

I still think that we don' have to distinguish between active and non-active workspaces though 🤔

Reply:
hmm maybe what he means is that we should just add a comment saying why are we loading `connections` here. We are loading them because if the policy is not the preferred (active) policy, then what we have loaded so far of the policy may be just a summary which doesn't contain the `connections` key.

Reply:
ahhh your explanation made it so clear Aldo! I'll edit the comment

Reply:
I've included a comment at the beginning, before the HOC definition, to explain the component's purpose.